crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2576,7 +2609,53 @@ impl<'db> TypeInferenceBuilder<'db> {
                         Type::Unknown
                     })
             }
-            _ => Type::Todo,
+            (value_ty, slice_ty) => {
+                // Resolve the value to its class.
+                let value_meta_ty = value_ty.to_meta_type(self.db);
+
+                // If the class defines `__getitem__`, return its return type.
+                //
+                // See: https://docs.python.org/3/reference/datamodel.html#class-getitem-versus-getitem
+                let dunder_getitem_method = value_meta_ty.member(self.db, "__getitem__");
+                if !dunder_getitem_method.is_unbound() {
+                    // TODO validate that the `__getitem__` argument adheres to the signature
```

Comment:
Is there anything to do here yet? E.g., I _could_ "manually" validate that `"abc"[()]` is not a valid operation, but I guess I'd rather get this "for free" from typeshed?

Reply:
Correct, nothing really to do here at this point!